modules/optimization/doc/content/modules/optimization/examples/shapeOpt_Annulus.md

```diff
@@ -0,0 +1,122 @@
+is an Augmented Lagrangian method. This is the only TAO solver in MOOSE that has
+constraint support. The `taoalmm` subsolver is set with [!param](/Executioner/Optimize/petsc_options_iname) and
+[!param](/Executioner/Optimize/petsc_options_value).
+constraint support. This example computes the objective function's gradient using finite differencing in TAO with the `petsc_options_iname` `-tao_fd`.  Finite differencing only uses the forward problem's objective value and does not require an adjoint solve.
+`-tao_almm_subsolver_tao_type` and  `bqnktr` set the subsolver to the Bounded Quasi-Newton-Krylov
+Trust Region Method.
+
+!listing test/tests/executioners/constrained/shape_optimization/main.i
+        block= Executioner
+        id=exec
+        caption= Main application `Executioner` block
+
+## Forward Problem Sub-App
+
+In the forward app, a heat conduction problem is solved with convective flux
+type boundary condition that changes with the inner radius. The outer radius is
+insulated and there is a constant source term throughout. The objective is
+find the two radii that will have the lowest max temperature while satisfying a
+volume constraint.
+
+!listing test/tests/executioners/constrained/shape_optimization/forward.i
+         id=forward_app
+         caption= Constrained Optimization Forward SubApp
+
+In the forward app, the two optimization parameters control the inner and outer
+values are used to displace the boundaries of the
+[ParsedOptimizationFunction](/Functions/ParsedOptimizationFunction) to calculate
+the cartesian boundary displacement. These
+values are used displace the boundaries of the
+annulus to the correct position using Dirichlet boundary conditions. A diffusion
+problem on the displacements then allow for the interior elements to be
+smoothed. For the temperature field, the physics is solved on the displaced mesh
+to calculate the current objective.
+
+For the equality constraints, a gradient of the constraint with respect to the
+parameters is needed for TAO. The gradient is analytically computed for the
+current radii.
+
+!listing test/tests/executioners/constrained/shape_optimization/forward.i
+         block= Functions BCs Kernels
+         id=forward_app_shape
+         caption= Forward App Shape Optimization
+
+Using postprocesors, the objective and constraint functions can be calculated.
+The objective in this case is to minimize the maximum temperature. Also the
+current volume is calculated and how much that volume violates the constraint.
+These values are now able to be transferred to the main application for the
+optimization process.
+
+
+!listing test/tests/executioners/constrained/shape_optimization/forward.i
+         block=Postprocessors
+         id=forward_app_obj
+         caption= Forward App Objective Calculations
+
+## Adjoint Problem Sub-App
+
+Since we are using finite-differencing to determine the gradients of our
+objective function, a Adjoint Sub-App is not needed.
```

Comment:
This is disappointing. Any reason we can't use the adjoint to compute the gradient?

Reply:
The objective is the maximum nodal temperature, which would not have a continuous gradient. Also one point of this example is to show that nearly any MOOSE app can be converted to a optimization problem using FD and a post processor.